modules/wayfire/hm.nix

```diff
@@ -1,66 +1,83 @@
 {
+  mkTarget,
   config,
   lib,
   pkgs,
   ...
 }:
-let
-  cfg = config.stylix.targets.wayfire;
-in
+mkTarget {
+  name = "wayfire";
+  humanName = "wayfire";
 
-{
-  options.stylix.targets.wayfire = {
-    enable = config.lib.stylix.mkEnableTarget "wayfire" true;
-    useWallpaper = config.lib.stylix.mkEnableWallpaper "wayfire" true;
-  };
-
-  config = lib.mkIf (config.stylix.enable && cfg.enable) (
-    let
-      inherit (config.lib.stylix) colors;
-      rgba = rgb: a: "\\#${rgb}${a}";
-      rgb = (lib.flip rgba) "ff";
+  extraOptions.useWallpaper = config.lib.stylix.mkEnableWallpaper "wayfire" true;
 
-      wayfireConfig = config.wayland.windowManager.wayfire;
+  configElements = [
+    (
+      { fonts }:
+      {
+        wayland.windowManager.wayfire.settings.decoration.font =
+          "${fonts.monospace.name} ${toString fonts.sizes.desktop}";
+      }
+    )
+    (
+      {
+        cfg,
+        image,
+        imageScalingMode,
+      }:
+      let
+        wayfireBackground = pkgs.runCommand "wayfire-background.png" { } ''
+          ${lib.getExe' pkgs.imagemagick "convert"} ${image} $out
+        '';
+      in
+      {
+        wayland.windowManager.wayfire.wf-shell.settings = {
+          background.image = lib.mkIf cfg.useWallpaper (toString wayfireBackground);
+          cube = {
+            cubemap_image = lib.mkIf cfg.useWallpaper (toString wayfireBackground);
+            skydome_texture = lib.mkIf cfg.useWallpaper (toString wayfireBackground);
+          };
+          background.fill_mode =
+            if imageScalingMode == "stretch" then
+              "stretch"
+            else if imageScalingMode == "fit" then
+              "preserve_aspect"
+            else
+              "fill_and_crop";
 
-      wayfireBackground = pkgs.runCommand "wayfire-background.png" { } ''
-        ${lib.getExe' pkgs.imagemagick "convert"} ${config.stylix.image} $out
-      '';
-    in
-    {
-      wayland.windowManager.wayfire.settings = lib.mkIf wayfireConfig.enable {
-        cube = {
-          background = rgb colors.base00;
-          cubemap_image = lib.mkIf cfg.useWallpaper (toString wayfireBackground);
-          skydome_texture = lib.mkIf cfg.useWallpaper (toString wayfireBackground);
+          panel.menu_icon = "${pkgs.nixos-icons}/share/icons/hicolor/256x256/apps/nix-snowflake.png";
```

Comment:
Despite this not depending on `image`, should it be guarded by `image`? Guarding this with `image`, implicitly allows end-users to opt out with `stylix.image = null;`, while never guarding this makes it impossible to implicitly opt out. Since this is not a value in the `stylix.image` sense, I am in favor of moving this into an unguarded block.